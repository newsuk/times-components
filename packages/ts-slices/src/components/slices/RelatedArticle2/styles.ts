import styled from 'styled-components';
import { breakpoints } from '@times-components/ts-styleguide';

import { calculateSlotWidth } from '../../../utils/getArticleStyles';

import {
  SliceContainer as SliceContainerBase,
  SlotContainer as SlotContainerBase
} from '../shared-styles';

export const SliceContainer = styled(SliceContainerBase)`
  @media (min-width: ${breakpoints.medium}px) {
    width: 80.8%;
    max-width: none;
    padding: 0;
  }

  @media (min-width: ${breakpoints.wide}px) {
    width: 56.2%;
  }
`;

export const SlotContainer = styled(SlotContainerBase)`
  @media (min-width: ${breakpoints.medium}px) {
    width: ${calculateSlotWidth(50, 2)};
    margin-bottom: 24px;
    padding-bottom: 24px;

    &:first-of-type {
      padding-right: 12px;

      :before {
        display: block;
        bottom: 24px;
      }
    }

    &:last-of-type {
      padding-left: 12px;
    }
  }
`;
