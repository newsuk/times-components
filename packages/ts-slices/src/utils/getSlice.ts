import React from 'react';

import { SliceData } from '../types/slice';
import { SliceStyle } from '../types/styles';
import { ClickHandlerType } from '../types/event';

import Lead1 from '../components/slices/Lead1/Lead1';
import Lead1And1 from '../components/slices/Lead1And1/Lead1And1';
import Lead1And2 from '../components/slices/Lead1And2/Lead1And2';
import Lead1And3Reversed from '../components/slices/Lead1And3Reversed/Lead1And3Reversed';
import Lead2 from '../components/slices/Lead2/Lead2';
import Secondary2And2 from '../components/slices/Secondary2And2/Secondary2And2';
import Secondary2And3 from '../components/slices/Secondary2And3/Secondary2And3';
import Secondary2And3NoPic from '../components/slices/Secondary2And3NoPic/Secondary2And3NoPic';
import Secondary3 from '../components/slices/Secondary3/Secondary3';
import Secondary4 from '../components/slices/Secondary4/Secondary4';
import Secondary4Odd from '../components/slices/Secondary4Odd/Secondary4Odd';
import Secondary10 from '../components/slices/Secondary10/Secondary10';
import RelatedArticle1 from '../components/slices/RelatedArticle1/RelatedArticle1';
import RelatedArticle2 from '../components/slices/RelatedArticle2/RelatedArticle2';
import RelatedArticle3 from '../components/slices/RelatedArticle3/RelatedArticle3';

export const getSliceComponent = (
  name: string
): React.FC<{
  slice: SliceData;
  styles?: SliceStyle;
  clickHandler?: ClickHandlerType;
}> => {
  switch (name) {
    case 'LEAD_1':
    case 'LEAD_1_COMPONENT':
    case 'INTERACTIVE_FULL_WIDTH':
      return Lead1;
    case 'LEAD_1_AND_1':
    case 'LEAD_1_AND_1_COMPONENT':
      return Lead1And1;
    case 'LEAD_1_AND_2':
    case 'LEAD_1_AND_2_COLUMNIST':
      return Lead1And2;
    case 'LEAD_1_AND_3_REVERSED':
      return Lead1And3Reversed;
    case 'LEAD_2':
    case 'LEAD_2_COMPONENT':
    case 'INTERACTIVE_HALF_WIDTH':
      return Lead2;
    case 'SECONDARY_2_AND_2':
      return Secondary2And2;
    case 'SECONDARY_2_AND_3':
      return Secondary2And3;
    case 'SECONDARY_2_AND_3_NO_PIC':
      return Secondary2And3NoPic;
    case 'SECONDARY_3':
      return Secondary3;
    case 'SECONDARY_4':
      return Secondary4;
    case 'SECONDARY_4_ODD':
    case 'SECONDARY_4_ODD_COLUMNIST':
      return Secondary4Odd;
    case 'SECONDARY_10':
      return Secondary10;
    case 'RELATED_ARTICLE_1':
      return RelatedArticle1;
    case 'RELATED_ARTICLE_2':
      return RelatedArticle2;
    case 'RELATED_ARTICLE_3':
      return RelatedArticle3;

    default:
      return () => null;
  }
};
