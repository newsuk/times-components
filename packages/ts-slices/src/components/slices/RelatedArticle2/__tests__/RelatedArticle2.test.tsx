import React from 'react';
import { render } from '@testing-library/react';

import { getSlice } from '../../../../fixtures/getSlice';

import RelatedArticle2 from '../RelatedArticle2';

jest.mock('../../../article/Article/Article', () => ({
  Article: () => <div>Article</div>
}));

describe('<RelatedArticle2 />', () => {
  afterAll(() => {
    jest.clearAllMocks();
  });

  it('should render an RELATED_ARTICLE_2 slice correctly', () => {
    const slice = getSlice('RELATED_ARTICLE_2');

    const { asFragment, getAllByText } = render(
      <RelatedArticle2 slice={slice} />
    );

    const article = getAllByText('Article');
    expect(article.length).toBe(2);

    expect(asFragment()).toMatchSnapshot();
  });
});
