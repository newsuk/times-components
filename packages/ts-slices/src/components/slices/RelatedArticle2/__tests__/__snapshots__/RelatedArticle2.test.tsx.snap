// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RelatedArticle2 /> should render an RELATED_ARTICLE_2 slice correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-bdVaJa sc-htpNat TayYz"
  >
    <div
      class="sc-bwzfXH sc-bxivhb ccFQOx"
    >
      <div>
        ArticleStandard
      </div>
    </div>
    <div
      class="sc-bwzfXH sc-bxivhb ccFQOx"
    >
      <div>
        ArticleStandard
      </div>
    </div>
  </div>
</DocumentFragment>
`;
